pkg/workqueue/workqueue.go

```diff
@@ -38,6 +40,24 @@ type WorkItem struct {
 	Callback func(ctx context.Context, obj any) error
 }
 
+// Return composite rate limiter that combines both per-item exponential backoff
+// and an overall token bucket rate-limiting strategy. It calculates the
+// exponential backoff for the individual item (based on its personal retry
+// history), checks the global rate against the token bucket, and picks the
+// longest delay from either strategy, ensuring that both per-item and overall
+// queue health are respected.
+func DefaultPrepUnprepRateLimiter() workqueue.TypedRateLimiter[any] {
```

Comment:
Can we call this `DefaultKubeletPluginRateLimiter` to be symmetrical with `DefaultControllerRateLimiter`.
Also, can you explain how this change is relevant to a PR focused on logging?

Reply:
> can you explain how this change is relevant to a PR focused on logging?

As many times, I kindly refer to my PR description :-)

https://github.com/NVIDIA/k8s-dra-driver-gpu/pull/633#issue-3469233142

As indicated in that PR description and also in another comment: I am happy to distribute the commits across different PRs to keep things separate.

Reply:
- took this part of the patch to https://github.com/NVIDIA/k8s-dra-driver-gpu/pull/656
- took this part of the discussion to https://github.com/NVIDIA/k8s-dra-driver-gpu/pull/656/files#r2413832932